api/src/main/java/org/apache/brooklyn/api/internal/AbstractBrooklynObjectSpec.java

```diff
@@ -65,7 +68,7 @@
     
     private final Class<? extends T> type;
     private String displayName;
-    private String catalogItemId;
```

Comment:
surely I take this into account already? See testRebindWithCatalogAndAppRebindCatalogItemIds,
and BrooklynMementoPersisterToObjectStore.getCatalogItemIds().

Could add it back but maybe don't need to?